jetty-jaas/src/main/java/org/eclipse/jetty/jaas/JAASLoginService.java

```diff
@@ -58,13 +56,14 @@
  * Implementation of jetty's LoginService that works with JAAS for
  * authorization and authentication.
  */
-public class JAASLoginService extends AbstractLifeCycle implements LoginService
+public class JAASLoginService extends ContainerLifeCycle implements LoginService
 {
     private static final Logger LOG = Log.getLogger(JAASLoginService.class);
 
     public static final String DEFAULT_ROLE_CLASS_NAME = "org.eclipse.jetty.jaas.JAASRole";
     public static final String[] DEFAULT_ROLE_CLASS_NAMES = {DEFAULT_ROLE_CLASS_NAME};
-
+    public static final ThreadLocal<JAASLoginService> INSTANCE = new ThreadLocal<>();
```

Comment:
This seems like an unusual way to share the `LoginService` with the `LoginModule`. 
Would it be better to do this with a `Callback` instead? similar how we would use `ServletRequestCallback`.

Reply:
The problem is that we allow the user to supply their own CallbackHandler class, which might not know anything about the special jetty  Callbacks, therefore couldn't supply a value for the JAASLoginService.

Reply:
Yeah that seems strange to me, how would you ever use `ServletRequestCallback` or `RequestParameterCallback` inside a login module because they only ever get set on the `DefaultCallbackHandler`? Seems like using this would also just break if they supplied a custom `CallbackHandler`.